From f1fd37c8dc7513d68bccc445553098095448bcec Mon Sep 17 00:00:00 2001
From: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Date: Tue, 25 Jun 2013 10:09:57 +0800
Subject: pwm: renesas-tpu: fix return value check in tpu_probe()

In case of error, the function devm_ioremap_resource() returns ERR_PTR()
and never returns NULL. The NULL test in the return value check should
be replaced with IS_ERR().

Signed-off-by: Wei Yongjun <yongjun_wei@trendmicro.com.cn>
Reviewed-by: Axel Lin <axel.lin@ingics.com>
Acked-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Thierry Reding <thierry.reding@gmail.com>
(cherry picked from commit 00cf99ee00c9f1241359c8ee5ca9230318e27a57)
Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
---
 drivers/pwm/pwm-renesas-tpu.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/pwm/pwm-renesas-tpu.c b/drivers/pwm/pwm-renesas-tpu.c
index 96e0cc48..03c1aa3c 100644
--- a/drivers/pwm/pwm-renesas-tpu.c
+++ b/drivers/pwm/pwm-renesas-tpu.c
@@ -410,10 +410,8 @@ static int tpu_probe(struct platform_device *pdev)
 	}
 
 	tpu->base = devm_ioremap_resource(&pdev->dev, res);
-	if (tpu->base == NULL) {
-		dev_err(&pdev->dev, "failed to remap I/O memory\n");
-		return -ENXIO;
-	}
+	if (IS_ERR(tpu->base))
+		return PTR_ERR(tpu->base);
 
 	tpu->clk = devm_clk_get(&pdev->dev, NULL);
 	if (IS_ERR(tpu->clk)) {
-- 
1.8.4.3.gca3854a

